packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -146,31 +226,46 @@ async function fetchCMRCollections({ collectionIds }) {
 }
 
 /**
- * Fetch collections in Elasticsearch.
- * @param {Object} recReportParams - input report params.
- * @returns {Promise<Array>} - list of collectionIds that match input paramaters
+ * Fetches collections from the database based on the provided parameters.
+ *
+ * @param {NormalizedRecReportParams} recReportParams - The reconciliation report parameters.
+ * @param {Knex} knex - The Knex.js database connection.
+ * @returns {Promise<string[]>} A promise that resolves to an array of collection IDs.
  */
-async function fetchESCollections(recReportParams) {
-  const esCollectionSearchParams = convertToESCollectionSearchParams(recReportParams);
-  const esGranuleSearchParams = convertToESGranuleSearchParams(recReportParams);
-  let esCollectionIds;
-  // [MHS, 09/02/2020] We are doing these two because we can't use
-  // aggregations on scrolls yet until we update elasticsearch version.
-  if (shouldAggregateGranulesForCollections(esGranuleSearchParams)) {
-    // Build an ESCollection and call the aggregateGranuleCollections to
-    // get list of collection ids that have granules that have been updated
-    const esCollection = new Collection({ queryStringParameters: esGranuleSearchParams }, 'collection', process.env.ES_INDEX);
-    const esCollectionItems = await esCollection.aggregateGranuleCollections();
-    esCollectionIds = esCollectionItems.sort();
-  } else {
-    // return all ES collections
-    const esCollection = new ESSearchQueue(esCollectionSearchParams, 'collection', process.env.ES_INDEX);
-    const esCollectionItems = await esCollection.empty();
-    esCollectionIds = esCollectionItems.map(
-      (item) => constructCollectionId(item.name, item.version)
-    ).sort();
+async function fetchDbCollections(recReportParams, knex) {
```

Comment:
There's probably a reasonable case to be made for this being generalized to the DB package, however both the typing and the specific modifiers pushed it over the fence for me *for now*, we should re-evaluate this when we take on the update/refactor work for recon reports.